sovrin_client/agent/jsonpickle_util.py

```diff
@@ -0,0 +1,64 @@
+import jsonpickle
+
+from anoncreds.protocol.types import PublicKey, RevocationPublicKey, \
+    SecretKey, RevocationSecretKey, AccumulatorSecretKey
+from anoncreds.protocol.utils import toDictWithStrValues, fromDictWithStrValues
+
+DATA_KEY = 'py/integer-element'
+
+
+class CommonIntegerElementHandler(jsonpickle.handlers.BaseHandler):
+    def flatten(self, obj, data):
+        data[DATA_KEY] = toDictWithStrValues(obj)
+        return data
+
+    def restore(self, obj):
+        dict = fromDictWithStrValues(obj[DATA_KEY])
+        return self._restore(dict)
+
+    def _restore(self, dict):
+        raise NotImplemented
+
+
+class PublicKeyHandler(CommonIntegerElementHandler):
```

Comment:
in all these handlers, when flatten (serialized) is called, I am calling **toDictWithStrValues** method and and when it gets restored (deserialized), I am calling **fromDictWithStrValues** (see CommonIntegerElementHandler class methods), and that way each element properly serialized and deserialized. There is test '**testAgentWalletRestoration**' which compares all the elements of the unpersisted walleted and restored wallet to make sure they are same.